waddrmgr/manager.go

```diff
@@ -1470,6 +1503,238 @@ func (m *Manager) LastInternalAddress(account uint32) (ManagedAddress, error) {
+		acctKeyPriv, err := deriveAccountKey(coinTypeKeyPriv, account)
+		if err != nil {
+			str := "failed to convert private key for account"
+			return managerError(ErrKeyChain, str, err)
+		}
+		acctKeyPub, err := acctKeyPriv.Neuter()
+		if err != nil {
+			str := "failed to convert public key for account"
+			return managerError(ErrKeyChain, str, err)
+		}
+		// Encrypt the default account keys with the associated crypto keys.
+		acctPubEnc, err := m.cryptoKeyPub.Encrypt([]byte(acctKeyPub.String()))
+		if err != nil {
+			str := "failed to  encrypt public key for account"
+			return managerError(ErrCrypto, str, err)
+		}
+		acctPrivEnc, err := m.cryptoKeyPriv.Encrypt([]byte(acctKeyPriv.String()))
+		if err != nil {
+			str := "failed to encrypt private key for account"
+			return managerError(ErrCrypto, str, err)
+		}
+		// We have the encrypted account extended keys, so save them to the
+		// database
+		err = putAccountInfo(tx, account, acctPubEnc, acctPrivEnc, 0, 0, name)
+		if err != nil {
+			return err
+		}
+		if err := putLastAccount(tx, account); err != nil {
+			return err
+		}
+		return putNumAccounts(tx, 1)
+	})
+	return account, err
+}
+
+// RenameAccount renames an account stored in the manager based on the
+// given account number with the given name.  If an account with the same name
+// already exists, ErrDuplicateAccount will be returned.
+func (m *Manager) RenameAccount(account uint32, name string) error {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	// Check that account with the new name does not exist
+	_, err := m.lookupAccount(name)
+	if err == nil {
+		str := fmt.Sprintf("account with the same name already exists")
+		return managerError(ErrDuplicateAccount, str, err)
+	}
+	var rowInterface interface{}
+	err = m.namespace.Update(func(tx walletdb.Tx) error {
+		var err error
+		rowInterface, err = fetchAccountInfo(tx, account)
+		if err != nil {
+			return err
+		}
+		// Ensure the account type is a BIP0044 account.
+		row, ok := rowInterface.(*dbBIP0044AccountRow)
+		if !ok {
+			str := fmt.Sprintf("unsupported account type %T", row)
+			err = managerError(ErrDatabase, str, nil)
+		}
+		// Remove the old name key from the accout id index
+		if err = deleteAccountIdIndex(tx, account); err != nil {
+			return err
+		}
+		// Remove the old name key from the accout name index
+		if err = deleteAccountNameIndex(tx, row.name); err != nil {
+			return err
+		}
+		err = putAccountInfo(tx, account, row.pubKeyEncrypted,
+			row.privKeyEncrypted, row.nextExternalIndex, row.nextInternalIndex, name)
+		return err
+	})
+	return err
+}
+
+// AccountName returns the account name for the given account number
+// stored in the manager.
+func (m *Manager) AccountName(account uint32) (string, error) {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	var acctName string
+	err := m.namespace.View(func(tx walletdb.Tx) error {
+		var err error
+		acctName, err = fetchAccountName(tx, account)
+		return err
+	})
+	if err != nil {
+		return acctName, err
```

Comment:
Please use explicit zero value returns since `acctName` could contain anything at this point.

```Go
return "", err
```